src/bsp/src/posix/src/SocketContainer.cpp

```diff
@@ -0,0 +1,22 @@
+#include "bsp/SocketContainer.h"
+#include "SocketFactory.h"
+#include "TCPClient.h"
+
+std::optional<TCPClientWrapper> SocketContainer::getHostClientSocket(const char* address,
+                                                                     int port,
+                                                                     const ILogger& logger) {
+
+    static std::optional<TCPClient> s_clientSocket = {};
+    static std::optional<TCPClientWrapper> s_wrapper = {};
+
+    if (!s_clientSocket) {
+        std::optional<TCPClient> socket = SocketFactory::createTCPClient(address, port, logger);
+
+        if (socket) {
+            s_clientSocket.emplace(socket.value());
```

Comment:
I suppose this is the awkward stuff for the `optional` you were talking about?

Reply:
Yep, that and the TCPWrapper stuff, since I can't return a std::optional of an incomplete type or a reference.
though a reference could be fixed with std::referece_wrapper, but still doesn't solve the incomplete type and it starts to get tedious typing